misc/config_tools/board_inspector/schema/checks/platform_capabilities.xsd

```diff
@@ -0,0 +1,288 @@
+      <xs:documentation>Enhanced rep movsb/stosb is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='invariant_tsc'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>Invariant TSC feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='tsc_deadline'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>TSC deadline feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='execute_disable'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>NX feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='smep'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>SMEP feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='smap'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>SMAP feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='mtrr'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>MTRR feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='clflushopt'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>CLFLUSHOPT feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='x2apic'])">
+    <xs:annotation acrn:severity="error">
+      <xs:documentation>x2APIC feature is not supported on this processor. ACRN requires this feature to function properly.</xs:documentation>
+    </xs:annotation>
+  </xs:assert>
+
+  <xs:assert test="every $model in processors/model satisfies exists($model/capability[@id='popcnt'])">
```

Comment:
How many of these specific feature tests do we need to look for?  Seems like if one of these isn't supported none of them will be, so there could be lots of errors written out (or none).

Reply:
Yes, it is by design to report every error log we detect, all these error logs will prevent user from booting ACRN.

Reply:
OK.